src/ploigos_step_runner/step_runner.py

```diff
@@ -168,12 +169,22 @@ def run_step(self, step_name, environment=None):
             self.workflow_result.add_step_result(
                 step_result=step_result
             )
-            self.workflow_result.write_to_pickle_file(
-                pickle_filename=self.workflow_result_pickle_file_path
-            )
-            self.workflow_result.write_results_to_yml_file(
-                yml_filename=self.results_file_path
-            )
+
+            # acquire the pickle lock
+            lock_filename = self.workflow_result_pickle_file_path + '.lock'
+            with open(lock_filename, 'w') as pickle_lock:
+                fcntl.flock(pickle_lock, fcntl.LOCK_EX)
```

Comment:
why not just lock on the pickle file, why create a seperate lock file?

Reply:
First instinct, mostly. I'll lock the pickle file for less complexity.